### PR TITLE
Treat compile warnings as errors

### PR DESCRIPTION
**Pull request by acj**
This should help to avoid silly bugs like the recent [@app_name](https://github.com/cjbottaro/faktory_worker_ex/pull/5#discussion_r159017525) issue.

### CODE REVIEW

File: test/support/stack.ex [hunk 1]
```diff
@@ -8,7 +8,7 @@ defmodule Stack do
     Agent.update(__MODULE__, &[item | &1])
   end
 
-  def pop(item) do
+  def pop do
```
Comment (acj) [inline test/support/stack.ex]: @cjbottaro This one is odd. The `Stack` module is referenced a handful of times, but I can't find anything that really uses it. The `Pinger` middleware module references `items` but doesn't push anything onto the stack, and `Pinger` itself doesn't seem to be referenced by anything. Is this dead code?

Comment (cjbottaro) [reply test/support/stack.ex]: Yeah, I think `Pinger` and `Stack` are dead code. They were originally meant to facilitate communication between a `Faktory.Job` and tests, but that's done differently now. I vote for removing them!

**Review by cjbottaro — CHANGES_REQUESTED** [review]
Can remove `Pinger` and `Stack` in this PR?

File: test/support/stack.ex (hunk 1 above)
Comment (acj) [reply test/support/stack.ex]: Sounds good. I should have time to finish this up tonight.

**Comment by acj** [comment]
Should be all set now.

**Pull request merged**